internal/exec/exec.go

```diff
@@ -0,0 +1,127 @@
+package exec
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"syscall"
+)
+
+type Execute struct {
+	StdOut   string
+	StdErr   string
+	ExitCode int
+}
+
+func isExecutable(path string) bool {
+	info, err := os.Stat(path)
+	if err != nil {
+		return false
+	}
+	return info.Mode()&0o111 != 0 // Check for any executable bit
+}
+
+func Run(executable string, environment map[string]string, args []string, stream bool, fileOutputStream string) (Execute, error) {
+	var result Execute
+
+	// Find execpath
+	execPath, err := exec.LookPath(executable)
+	if err != nil {
+		return result, fmt.Errorf("executable not found or not in PATH: %s", executable)
+	}
+	if !isExecutable(execPath) {
+		return result, fmt.Errorf("executable is not permitted to run: %s", execPath)
+	}
+
+	// Set up command
+	cmd := exec.Command(execPath, strings.Join(args, " "))
+
```

Comment:
Maybe use CommandContext so we can cancel? I think @terrabyteian  would know

Reply:
swapped it over to accept context now. just added a basic one for the actual cobra-cli call. dont really anticipate that being the file function. its more or less to test out the functionality, but i think we should maybe support a base shell call for debugging or anything like that.